### ISSUE TITLE
Simplify canvas settings window

### ISSUE THREAD

**Issue by morevnaproject** [synfig/synfig#777]
I am proposing to create a simplified Canvas settings window, which will allow to easily set dimensions of working document. 

At this moment Canvas Properties window looks like this (many options and 3 tabs!):
![screenshot_018](https://user-images.githubusercontent.com/332868/54493291-beb6f600-4900-11e9-8910-c4d7d7c512f4.png)

We need something like this (taken from Flash):

![flash_s](https://user-images.githubusercontent.com/332868/54493285-9c24dd00-4900-11e9-877e-553432f66117.png)

Useful reading on this topic: http://www.segtsy.com/2013/12/a-significant-synfig-setback/

**Comment by rockyhandsome1** [comment]
Do you mean that the entire properties window to be replaced by the documents settings window. If yes then the user will be left with just few options..

**Comment by morevnaproject** [comment]
@rockyhandsome1 My suggestion to create a simple window and add "Advanced..." button there. Clicking it will display the current window with all complicated options.

**Comment by morevnaproject** [comment]
So, I assume we need to cover following cases:

1. **"Canvas" -> "Resize"** - Resizing canvas together with its content.
By default width linked with height to keep aspect ratio. User can unlink those values and in this case canvas expands appropriately.
This is current default behavior of "Canvas->Properties" window with Locks and Links set as follows:
![screenshot_007](https://user-images.githubusercontent.com/332868/55499956-66f2eb80-5671-11e9-92d8-3c1989455085.png)
2. **"Canvas" -> "Set Dimensions"** - Expanding/shrinking canvas (without changing its content size). 
By default width and height do not depend on each other. But user can link them to keep aspect ratio,
This is an equivalent of following settings for Locks and Links in "Canvas->Properties" window:
![screenshot_008](https://user-images.githubusercontent.com/332868/55500205-07491000-5672-11e9-8644-b273aa728498.png)
3. **"File" -> "Render"** - Set size for rendering output. Resizes image together with its content.
By default width linked with height to keep aspect ratio. User can unlink those values and in this case content of canvas is stretched as shown in https://github.com/synfig/synfig/issues/771#issuecomment-478359716
This is an equivalent of following settings for Locks and Links in "Canvas->Properties" window:
![screenshot_009](https://user-images.githubusercontent.com/332868/55500292-38294500-5672-11e9-871d-1f8697b44a33.png)

**Comment by bajajvinamr** [comment]
Is this issue open at the moment and if yes, i would like to work upon it

**Comment by morevnaproject** [comment]
Hi @bajajvinamr and welcome to Synfig! Your help is much appreciated here. ^__^

BTW< I received your message about course request, which you sent via contact form. But I wasn't able to reply - message returns with a "No Such User Here" error.
Possibly mistake in your contact email address? Please send one more message with correct email via contact form again. ^__^

**Comment by ankit-kumar-dwivedi** [comment]
@bajajvinamr assigning you this one :)

**Comment by ice0** [comment]
@bajajvinamr do you want to continue working on this?

**Comment by rodolforg** [comment]
@morevnaproject  do you mean simple menu entries?

>     1. **"Canvas" -> "Resize"**



>     2\. **"Canvas" -> "Set Dimensions"** -

**Comment by chandratushu21** [comment]
I would like to work on this ussue.

**Comment by rodolforg** [comment]
@chandratushu21 your help would be appreciated ;)

**Comment by chandratushu21** [comment]
How can I start working on this one ?

**Comment by morevnaproject** [comment]
> @morevnaproject do you mean simple menu entries?
> 
> > ```
> > 1. **"Canvas" -> "Resize"**
> > ```
> 
> > ```
> > 2. **"Canvas" -> "Set Dimensions"** -
> > ```

Apologies for such a long delay. Yes, this is exactly what I mean. ^__^

...and for each menu entry show a simple dialog, where user can set dimensions (in pixels).

![screenshot_006](https://user-images.githubusercontent.com/332868/84597013-243ebd00-ae8b-11ea-8ac3-90385fd3199b.png)

@chandratushu21 Let's get started! Did you managed to build Synfig from source already? ^__^

**Comment by chandratushu21** [comment]
@morevnaproject don’t know how to go with that sir, I'm appearing for the first time. Can you help for this issue or suggest some other easy issue to understand the workflow.
Thanks 🙏

**Comment by morevnaproject** [comment]
@chandratushu21 Sure! Please check instructions on how to build Synfig from source - https://synfig-docs-dev.readthedocs.io/en/latest/common/building.html

**Comment by chandratushu21** [comment]
> @chandratushu21 Sure! Please check instructions on how to build Synfig from source - https://synfig-docs-dev.readthedocs.io/en/latest/common/building.html

@morevnaproject Sir I followed up and successfully checked prelims but found the following error in the terminal while executing **"2-build-debug.sh”** under first build, ref;  @https://synfig-docs-dev.readthedocs.io/en/latest/common/building.html

<img width="480" alt="Screenshot 2020-06-16 at 10 13 18 PM" src="https://user-images.githubusercontent.com/64960756/84803052-1326b500-b01f-11ea-904a-e554ce98e18e.png">

Please help.
Thanks 🙏

**Comment by ice0** [comment]
You can see how Rodolfo solved this problem here (https://github.com/synfig/synfig/pull/1486)

**Comment by chandratushu21** [comment]
@ice0  Thanks sir, 
Well I haven't done this before so is there is any risk to harm my machine by doing so without proper knowledge and what if I want to remove this project ? For ex. homebrew has installed many things that I don't know so do I need to remove them or they will be occupied by every project or so.

**Comment by chandratushu21** [comment]
@morevnaproject Sir done with building synfig from source and the program is running well. I'm very happy on my first achievement, 
BTW, what next ? 
Where can I start resolving the issue of canvas properties.

**Comment by ice0** [comment]
> Well I haven't done this before so is there is any risk to harm my machine by doing so without proper knowledge and what if I want to remove this project ? For ex. homebrew has installed many things that I don't know so do I need to remove them or they will be occupied by every project or so.

You can remove synfig folder and brew (if you are not using it for something else). This will not harm your computer.
https://docs.brew.sh/FAQ

**Comment by morevnaproject** [comment]
> BTW, what next ?
> Where can I start resolving the issue of canvas properties.

Great! Now I suggest you to edit code of `synfig-studio` and add new menu items (at first they will do nothing). You need to add those menu items:
- **"Canvas" -> "Resize"**
- **"Canvas" -> "Set Dimensions"**

**Comment by chandratushu21** [comment]
@morevnaproject Roger that sir ! On it.. 😊 
First trying to study and figure out code.

**Comment by ankit-kumar-dwivedi** [comment]
@chandratushu21 are you still working on this?

**Comment by chandratushu21** [comment]
@ankit-kumar-dwivedi Not now, Tried to but failed so switched to some other projects. Sorry didn’t informed earlier. Thanks   for the help @morevnaproject @ice0

**Comment by Keyikedalube** [comment]
> So, I assume we need to cover following cases:
> 
> 1. **"Canvas" -> "Resize"** - Resizing canvas together with its content.
>     By default width linked with height to keep aspect ratio. User can unlink those values and in this case canvas expands appropriately.
>     This is current default behavior of "Canvas->Properties" window with Locks and Links set as follows:
>     ![screenshot_007](https://user-images.githubusercontent.com/332868/55499956-66f2eb80-5671-11e9-92d8-3c1989455085.png)

Just **Image Width** and **Image Height** turned on and the rest off? Or along with **Pixel Aspect** and **Image Span**?

> 2. **"Canvas" -> "Set Dimensions"** - Expanding/shrinking canvas (without changing its content size).
>     By default width and height do not depend on each other. But user can link them to keep aspect ratio,
>     This is an equivalent of following settings for Locks and Links in "Canvas->Properties" window:
>     ![screenshot_008](https://user-images.githubusercontent.com/332868/55500205-07491000-5672-11e9-8644-b273aa728498.png)

Does that mean **Locks and Links** will have to be there in each dialog ie., **Resize** and **Set Dimensions**?

Also, is **Locks and Links** safe to be removed from **Canvas Properties** dialog?

I might want to take on this issue after #1818 is resolved.

**Comment by morevnaproject** [comment]
> > So, I assume we need to cover following cases:
> > 
> > 1. **"Canvas" -> "Resize"** - Resizing canvas together with its content.
> >    By default width linked with height to keep aspect ratio. User can unlink those values and in this case canvas expands appropriately.
> >    This is current default behavior of "Canvas->Properties" window with Locks and Links set as follows:
> >    ![screenshot_007](https://user-images.githubusercontent.com/332868/55499956-66f2eb80-5671-11e9-92d8-3c1989455085.png)
> 
> Just **Image Width** and **Image Height** turned on and the rest off? Or along with **Pixel Aspect** and **Image Span**?

**Image Width** and **Image Height** should be turned off.
The linking of width with height is achieved through different button on "Image" tab -
![screenshot_002](https://user-images.githubusercontent.com/332868/99978757-ef8b4600-2dd8-11eb-81dc-422ccad7c1d7.png)
AFAIK it doesn't uses **Image Width** and **Image Height** flags...


**Pixel Aspect** and **Image Span** should be on, and the rest options are off.

**Comment by morevnaproject** [comment]
> Does that mean **Locks and Links** will have to be there in each dialog ie., **Resize** and **Set Dimensions**?

I assume **Locks and Links** shouldn't be visible. Those are internal parameters, not needed by average user.

I expect this feature to be implemented as follows:

1. Add **"Canvas" -> "Resize..."** menu entry.
2. This menu entry will show a simple dialog with those simple elements -
![screenshot_004](https://user-images.githubusercontent.com/332868/99979271-7cce9a80-2dd9-11eb-8a1f-f2536e74ab20.png)
3. Dialog will also include a button with "Advanced..." text label + OK/cancel buttons.
4. Clicking "Advanced..." button hiddes the simple dialog and shows complicated "Properties" window (the one which is currently shown when we call "Canvas" -> "Properties" menu item).
5. When user clicks OK. the code takes Width/Height values and passing proper **Locks and Links** flags to RendDesc (see https://github.com/synfig/synfig/issues/771#issuecomment-479579591 for details).

Same for Canvas resize function:

1. Add **"Canvas" -> "Set Dimensions..."** menu entry.
2. This menu entry will show a simple dialog with those simple elements -
![screenshot_003](https://user-images.githubusercontent.com/332868/99979929-3f1e4180-2dda-11eb-9a28-18bce6619b75.png)
3. Dialog will also include a button with "Advanced..." text label + OK/cancel buttons.

**Comment by Keyikedalube** [comment]
Does that mean...
1. Width, Height, and link toggle-button (your screenshots above) have to be removed from the **Canvas Properties** dialog?
2. Remove Canvas Properties menu item from the menubar? And let **Resize** and **Set Dimensions** "Advanced..." button open the said dialog?

**Comment by morevnaproject** [comment]
>     1. Width, Height, and link toggle-button (your screenshots above) have to be removed from the **Canvas Properties** dialog?

No, I think it is okay to keep it.


>     2. Remove Canvas Properties menu item from the menubar? And let **Resize** and **Set Dimensions** "Advanced..." button open the said dialog?

Yes, "Canvas" - "Properties" can be removed, but as the last step (when **Resize** and **Set Dimensions** will be fully implemented).

**Comment by Keyikedalube** [comment]
@morevnaproject Got it and thanks :)

**Comment by Keyikedalube** [comment]
BTW, Canvas Properties dialog should also be redesigned. I made a PR a while ago where I implemented Canvas Properties dialog glade file and adjusted the main notebook content alignment for UI consistency. Then I closed it and mentioned the reason https://github.com/synfig/synfig/pull/1577#issuecomment-719881157 for the new dialog window being too large. In the screenshot, the widgets from Image tab may look OK but not the other tabs. Body content was too spacious and also notice the large **Name** and **Description** entry widget. Overall, the current design became more inconsistent with space distribution.

I have this new Image tab layout in mind but it's not the best either:
```
Width  <spin-entry> t    Physical Width  <spin-entry>
Height <spin-entry> b    Physical Height <spin-entry>

XRes   <spin-entry> t          Image Size Ratio
YRes   <spin-entry> b              <ratio>
```

**Pro** - Large width length is adjusted at least
**Con** - Height is gained

**Also a silly pro** - I get to reuse, rebase, and re-submit my PR again but under a new branch name LOL

**Comment by Keyikedalube** [comment]
> Useful reading on this topic: http://www.segtsy.com/2013/12/a-significant-synfig-setback/

Read that post. I don't heavily work on imported raster files so I'm oblivious to this nasty pothole. I mostly prefer drawing from Inkscape and saving it to synfig file format afterward.

The blogger clearly explained the weirdness between
- what's seen (thought canvas was resized too) and
- what's happening (was resized only for output rendering)

and I agree with the post. We're not supposed to expect users to understand Synfig specific internals. That's bad design.

I assume **Image Area** widgets values have to be automatically computed from **Image Size** Width and Height values? And safely remove the former section from Canvas Properties dialog.

**Comment by Keyikedalube** [comment]
> I have this new Image tab layout in mind but it's not the best either:
>```
>Width  <spin-entry> t    Physical Width  <spin-entry>
>Height <spin-entry> b    Physical Height <spin-entry>
>
>XRes   <spin-entry> t          Image Size Ratio
>YRes   <spin-entry> b              <ratio>
>```

Or put the above new layout Image Size section in a new tab called **Render** and rename **Image** tab to **Canvas**

**Comment by Keyikedalube** [comment]
I'm working on this issue :)

**Comment by rodolforg** [comment]
Well, Synfig actually

> I assume **Image Area** widgets values have to be automatically computed from **Image Size** Width and Height values? And safely remove the former section from Canvas Properties dialog.

In normal cases, yes. But some advanced users don't like removing this feature.

Konstantin also proposed in other issue to provide simple/direct canvas (re)sizes IARC

**Comment by Keyikedalube** [comment]
Anyway, I'm not touching Canvas Properties dialog. My suggestions/proposals would only end up making the dialog more complicated with too many tabs and isolated widgets that depend on each other.

I find the names **Image Size** and **Image Area** in the Image tab quite confusing. Should be named something like **Render Size** and **Canvas Size**. Size and Area quite denote the same thing but with names like Render and Canvas... we now know those widgets specific purpose. But this is a separate issue so not fixing it here.

**Comment by morevnaproject** [comment]
> I assume **Image Area** widgets values have to be automatically computed from **Image Size** Width and Height values? And safely remove the former section from Canvas Properties dialog.

I think the purpose of those widgets is to set the placement of (0,0) point? ^__^
With default values it is in the middle of canvas.
![screenshot_003](https://user-images.githubusercontent.com/332868/101001596-e5fc9d80-3591-11eb-8436-1cb400873cca.png)

Maybe some users might want to have (0,0) point in top-left corner.
But in this case (in my opinion) we have to provide a different GUI approach to set this (maybe by right-clicking a ruler?)
BTW, I am curious how this solved in AfterEffects?

**Comment by Keyikedalube** [comment]
Thanks for the info. A while ago I read this [Canvas Properties wiki](https://wiki.synfig.org/Canvas_Properties_Dialog#Image_tab) page. Those widgets (and core synfig code) are too complicated for me. I don't really know their specific use cases other than guessing those are for advanced users who wishes to have absolute or just more control of their canvas ;P

> BTW, I am curious how this solved in AfterEffects?

Sorry, not my area :(

My focus is fixed on what you laid above: **Canvas Resize** and **Set Dimensions** dialogs. Canvas Properties remain intact.
